app/src/main/res/layout/fragment_move.xml

```diff
@@ -69,48 +69,87 @@
-                    android:layout_height="wrap_content"
-                    android:nestedScrollingEnabled="false"
-                    android:overScrollMode="never"
-                    app:layoutManager="androidx.recyclerview.widget.LinearLayoutManager"
-                    tools:itemCount="6"
-                    tools:listitem="@layout/item_folder_menu_drawer" />
+                    <LinearLayout
+                        android:id="@+id/moveFoldersLists"
+                        android:layout_width="match_parent"
+                        android:layout_height="match_parent"
+                        android:animateLayoutChanges="true"
+                        android:orientation="vertical">
 
-            </LinearLayout>
-        </androidx.core.widget.NestedScrollView>
+                        <androidx.recyclerview.widget.RecyclerView
+                            android:id="@+id/defaultFoldersList"
+                            android:layout_width="match_parent"
+                            android:layout_height="wrap_content"
+                            android:nestedScrollingEnabled="false"
+                            android:overScrollMode="never"
+                            app:layoutManager="androidx.recyclerview.widget.LinearLayoutManager"
+                            tools:itemCount="4"
+                            tools:listitem="@layout/item_folder_menu_drawer" />
+
+                        <com.google.android.material.divider.MaterialDivider
+                            style="@style/MenuDrawerDivider"
+                            android:layout_width="match_parent"
+                            android:layout_height="wrap_content" />
+
+                        <androidx.recyclerview.widget.RecyclerView
+                            android:id="@+id/customFoldersList"
+                            android:layout_width="match_parent"
+                            android:layout_height="wrap_content"
+                            android:nestedScrollingEnabled="false"
+                            android:overScrollMode="never"
+                            app:layoutManager="androidx.recyclerview.widget.LinearLayoutManager"
+                            tools:itemCount="6"
+                            tools:listitem="@layout/item_folder_menu_drawer" />
+                    </LinearLayout>
+
+                    <androidx.recyclerview.widget.RecyclerView
+                        android:id="@+id/searchResultsList"
+                        android:layout_width="match_parent"
+                        android:layout_height="match_parent"
+                        android:nestedScrollingEnabled="false"
+                        android:overScrollMode="never"
+                        android:visibility="gone"
+                        app:layoutManager="androidx.recyclerview.widget.LinearLayoutManager"
+                        tools:itemCount="6"
+                        tools:listitem="@layout/item_folder_menu_drawer" />
```

Comment:
It's a good idea to put view that cannot be displayed at the same time inside a FrameLayout but here this implies that this new RecyclerView won't recycle because of the parent NestedScrollView. Because the RecyclerView can already scroll on its own we should at least take that one out of the NestedScrollView. You can then have a FrameLayout holding two childs: the nested and this recycler

Reply:
Indeed